samples/aws-samples/mnist-kmeans-sagemaker/kmeans-hpo-pipeline.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import kfp
+from kfp import components
+from kfp import dsl
+from kfp.aws import use_aws_secret
+
+sagemaker_hpo_op = components.load_component_from_file('../../../components/aws/sagemaker/hyperparameter_tuning/component.yaml')
+
+@dsl.pipeline(
+    name='MNIST HPO test pipeline',
+    description='SageMaker hyperparameter tuning job test'
+)
+def hpo_test(region='us-west-2',
+    hpo_job_name='HPO-kmeans-sample',
+    image='',
+    algorithm_name='K-Means',
+    training_input_mode='File',
+    metric_definitions='{}',
+    strategy='Bayesian',
+    metric_name='test:msd',
+    metric_type='Minimize',
+    early_stopping_type='Off',
+    static_parameters='{"k": "10", "feature_dim": "784"}',
+    integer_parameters='[{"Name": "mini_batch_size", "MinValue": "450", "MaxValue": "550"}, \
+                         {"Name": "extra_center_factor", "MinValue": "10", "MaxValue": "20"}]',
+    continuous_parameters='[]',
+    categorical_parameters='[{"Name": "init_method", "Values": ["random", "kmeans++"]}]',
+    channels='[{"ChannelName": "train", \
+                "DataSource": { \
+                    "S3DataSource": { \
+                        "S3Uri": "s3://kubeflow-pipeline-data/mnist_kmeans_example/data",  \
+                        "S3DataType": "S3Prefix", \
+                        "S3DataDistributionType": "FullyReplicated" \
+                        } \
+                    }, \
+                "ContentType": "", \
+                "CompressionType": "None", \
+                "RecordWrapperType": "None", \
+                "InputMode": "File"}, \
+               {"ChannelName": "test", \
+                "DataSource": { \
+                    "S3DataSource": { \
+                        "S3Uri": "s3://kubeflow-pipeline-data/mnist_kmeans_example/data", \
+                        "S3DataType": "S3Prefix", \
+                        "S3DataDistributionType": "FullyReplicated" \
+                        } \
+                    }, \
+                "ContentType": "", \
+                "CompressionType": "None", \
+                "RecordWrapperType": "None", \
+                "InputMode": "File"}]',
+    output_location='s3://kubeflow-pipeline-data/mnist_kmeans_example/output',
+    output_encryption_key='',
+    instance_type='ml.p2.16xlarge',
+    instance_count='1',
+    volume_size='50',
+    max_num_jobs='1',
+    max_parallel_jobs='1',
+    resource_encryption_key='',
+    max_run_time='3600',
+    vpc_security_group_ids='',
+    vpc_subnets='',
+    network_isolation='True',
+    traffic_encryption='False',
+    warm_start_type='',
+    parent_hpo_jobs='',
+    tags='{}',
+    role_arn='',
+    ):
+
+    training = sagemaker_hpo_op(
```

Comment:
Basically, does user create SM HOP job separately? Do they pipeline this job with other training jobs or something else?

Reply:
The user can do it separately like in this sample, but they can also pipeline this job with another training or create model component since it outputs the tuned hyperparameters and the job name and model artifacts url of the best training job within the HPO job. I'll have an example with it in a pipeline with other components

Reply:
What I am thinking is it's better to have a more complex example to demo real world use case like taxi example.

Reply:
Okay, will work on making a more complex example when I get a chance to